app/src/main/java/waiter/App.java

```diff
@@ -3,12 +3,16 @@
  */
 package waiter;
 
+import java.io.IOException;
+import java.net.ServerSocket;
+
 public class App {
-    public String getGreeting() {
-        return "Hello World!";
-    }
 
-    public static void main(String[] args) {
-        System.out.println(new App().getGreeting());
+    public static void main(String[] args) throws IOException {
```

Comment:
I'm not sure we want to throw here. Do we want the server to blow up upon starting up or do we want to handle any exceptions gracefully?

Reply:
Hmmm, Im not quite sure, but I would think we would want to handle it gracefully, would this be done via something like a try catch?

Reply:
Yea I think try catch should do it. At any given time we don't want our application to crash. In case where the problem can't be resolved and the user needs to re-run the application, we should let the user know that instead of just throwing an exception.

Reply:
I think that's ok for now. Ideally we can catch a SocketExeception as well as an IOException so that we can distinguish between a connection issue or an IO issue(reading from or writing to the stream). That also means instead of throwing IOException in the `Listener`, we would throw a SocketException instead.

Reply:
That does make sense, I will put in a basic try catch with a "There was a problem starting the echo server, please check your command and rerun" for now, but Ill create another PR in the future to address this too! Thank you for the explanation as well 😄

Reply:
Sounds good!

Reply:
Resolved in [this commit](https://github.com/jonmanahan/waiter/pull/2/commits/bf40a8e50470f025810a180272a6c38774729300), but also want to add throwing SocketExceptions in another PR as discussed above. Potential ticket name (for tracking purposes) -> add-socket-exception-catch Ticket, thank you for all the help, you guys are awesome!